src/qibo/quantum_info/linalg_operations.py

```diff
@@ -196,3 +196,28 @@ def matrix_exponentiation(
     backend = _check_backend(backend)
 
     return backend.calculate_matrix_exp(phase, matrix, eigenvectors, eigenvalues)
+
+
+def matrix_power(matrix, power: Union[float, int], backend=None):
+    """Given a ``matrix`` :math:`A` and power :math:`\\alpha`, calculates :math:`A^{\\alpha}`.
+
+    Args:
+        matrix (ndarray): matrix whose power to calculate.
+        power (float or int): power to raise ``matrix`` to.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend
+            to be used in the execution. If ``None``, it uses
+            :class:`qibo.backends.GlobalBackend`. Defaults to ``None``.
+
+    Returns:
+        ndarray: matrix power :math:`A^{\\alpha}`.
+    """
+
+    if not isinstance(power, (float, int)):
+        raise_error(
+            TypeError,
+            f"``power`` must be either float or int, but it is type {type(power)}.",
+        )
```

Comment:
If we decide to keep this check, I would keep it only in the backend function, I don't think you need redundancy here.

Reply:
This goes in the same basket as the refactoring comment of the other PR

Reply:
Besides the refactor, here you have the same identical check that happens in two functions that are concatenated:
``` python
def internal_function(...):
    check(...)
    ...

def external_f(...):
    check(...)
    return internal_function(...)
```
the check will always happen even if you remove it from the external function, i.e. this one here.